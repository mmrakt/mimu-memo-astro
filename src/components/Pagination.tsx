import React from 'react'

import ChevronLeft from './common/ChevronLeft'
import ChevronRight from './common/ChevronRight'

import type { Page } from 'astro'

type IProps = {
  page: Pick<
    Page,
    'start' | 'end' | 'size' | 'total' | 'currentPage' | 'lastPage' | 'url'
  >
}

export const Pagination: React.FC<IProps> = ({ page }) => {
  const { start, end, size, total, currentPage, lastPage, url } = page

  return (
    <div className="flex py-3">
      {currentPage !== 1 && (
        <a
          href={url.prev}
          rel="prev"
          className="box-outline text-link flex items-center pr-2 py-2 dark:text-white underline text-sm font-medium dark:bg-dark-gray bg-white"
        >
          <ChevronLeft />
          <span className="">Prev</span>
        </a>
      )}
      {currentPage !== lastPage && (
        <a
          href={url.next}
          rel="next"
          className="box-outline text-link flex items-center ml-auto pl-2 py-2 dark:text-white underline text-sm font-medium dark:bg-dark-gray bg-white"
        >
          <span>Next</span>
          <ChevronRight />
        </a>
      )}
    </div>
  )
}

export default Pagination
