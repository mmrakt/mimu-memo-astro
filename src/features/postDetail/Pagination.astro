---
import useStringTrim from '../../hooks/useStringTrim'

import type { PaginatedPost } from '../../types'

type Props = {
  next: PaginatedPost['next']
  prev: PaginatedPost['prev']
}
const { next, prev } = Astro.props
---

<div class="postDetailPagination">
  <div class="postDetailPagination__item">
    {
      prev.url && (
        <a
          href={`/${prev.url}`}
          class="postDetailPagination__itemLink postDetailPagination__itemLink--prev"
        >
          {useStringTrim(prev.title, 50)}
        </a>
      )
    }
  </div>
  <div class="postDetailPagination__item">
    {
      next.url && (
        <a
          href={`/${next.url}`}
          class="postDetailPagination__itemLink postDetailPagination__itemLink--next"
        >
          {useStringTrim(next.title, 50)}
        </a>
      )
    }
  </div>
</div>

<style lang="scss">
  @use '../../styles/utils' as utils;
  @use '../../styles/variables' as vars;
  .postDetailPagination {
    display: flex;
    gap: 20px;
    align-items: center;
    margin-top: 40px;

    &__item {
      width: 50%;
    }
    &__itemLink {
      @extend %link;
      display: flex;
      align-items: center;
      position: relative;

      &--prev {
        padding-left: 20px;
        &::before {
          @extend %arrow-prev;
        }
        &:hover {
          &::before {
            border-color: vars.$black-400;
          }
        }
        @include utils.dark() {
          &:hover {
            &::before {
              border-color: vars.$black-700;
            }
          }
        }
      }
      &--next {
        padding-right: 20px;
        &::after {
          @extend %arrow-next;
        }
        &:hover {
          &::after {
            border-color: vars.$black-400;
          }
        }
        @include utils.dark() {
          &:hover {
            &::after {
              border-color: vars.$black-700;
            }
          }
        }
      }
    }
  }
</style>
