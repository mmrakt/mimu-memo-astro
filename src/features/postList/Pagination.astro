---
import type { Page } from 'astro'

type Props = {
  page: Pick<Page, 'currentPage' | 'lastPage' | 'url'>
}
const {
  page: { currentPage, lastPage, url },
} = Astro.props
---

<div class="postListPagination flex py-3">
  {
    currentPage !== 1 && (
      <a
        href={url.prev}
        rel="prev"
        class="postListPagination__link postListPagination__link--prev"
      >
        Prev
      </a>
    )
  }
  {
    currentPage !== lastPage && (
      <a
        href={url.next}
        rel="next"
        class="postListPagination__link postListPagination__link--next"
      >
        Next
      </a>
    )
  }
</div>

<style lang="scss">
  @use '../../styles/utils' as utils;
  @use '../../styles/variables' as vars;

  .postListPagination {
    display: flex;
    margin-top: 40px;

    &__link {
      display: flex;
      align-items: center;
      justify-content: center;
      padding: 8px 12px;
      font-weight: bold;
      position: relative;

      @extend %link;
      @extend %card-content;

      &--prev {
        padding-left: 30px;
        &::before {
          @extend %arrow-prev;
          left: 10px;
        }

        &:hover {
          &::before {
            border-color: vars.$black-400;
          }
        }
        @include utils.dark() {
          &:hover {
            &::before {
              border-color: vars.$black-700;
            }
          }
        }
      }

      &--next {
        padding-right: 30px;
        margin-left: auto;

        &::after {
          @extend %arrow-next;
          right: 10px;
        }

        &:hover {
          &::after {
            border-color: vars.$black-400;
          }
        }
        @include utils.dark() {
          &:hover {
            &::after {
              border-color: vars.$black-700;
            }
          }
        }
      }
    }
  }
</style>
