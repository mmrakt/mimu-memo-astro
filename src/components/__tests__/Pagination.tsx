import { render } from '@testing-library/react'
import * as React from 'react'

import Pagination from '../Pagination'

// TODO: [Consider using the "jsdom" test environment.] のエラーの対応
it.skip('renders correctly', () => {
  const pageContext = {
    previousPagePath: '/',
    nextPagePath: '/page/3',
  }
  const { container } = render(<Pagination pageContext={pageContext} />)
  expect(container).toMatchSnapshot()
})
