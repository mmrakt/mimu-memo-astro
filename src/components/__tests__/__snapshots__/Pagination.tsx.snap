// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex py-3"
>
  <a
    className="box-outline text-link flex items-center pr-2 py-2 dark:text-white underline text-sm font-medium dark:bg-dark-gray bg-white"
    href="/"
    rel="prev"
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M15.41 7.41L14 6l-6 6 6 6 1.41-1.41L10.83 12z"
      />
    </svg>
    <span
      className=""
    >
      Prev
    </span>
  </a>
  <a
    className="box-outline text-link flex items-center ml-auto pl-2 py-2 dark:text-white underline text-sm font-medium dark:bg-dark-gray bg-white"
    href="/page/3"
    rel="next"
  >
    <span>
      Next
    </span>
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 6L8.59 7.41 13.17 12l-4.58 4.59L10 18l6-6z"
      />
    </svg>
  </a>
</div>
`;
